dired-collapse.el

```diff
@@ -127,27 +125,30 @@ filename (for example when the final directory is empty)."
 (defun dired-collapse ()
   "Collapse unique nested paths in dired listing."
   (when (or (not (file-remote-p default-directory)) dired-collapse-remote)
-    (-let* (;; dired-hide-details-mode hides details by assigning a special invisibility text property
-            ;; to them, while dired-collapse requires all the details. So we disable invisibility here
-            ;; temporarily.
-            (buffer-invisibility-spec nil)
-            (inhibit-read-only t))
+    (let (;; dired-hide-details-mode hides details by assigning a
+          ;; special invisibility text property to them, while
+          ;; dired-collapse requires all the details. So we disable
+          ;; invisibility here temporarily.
+          (buffer-invisibility-spec nil)
+          (inhibit-read-only t))
       (save-excursion
         (goto-char (point-min))
         (while (not (eobp))
-          (when-let ((filename-no-dir (dired-utils-get-filename 'no-dir)))
+          (when-let ((filename-no-dir (dired-get-filename 'no-dir t)))
             (when (and (looking-at-p dired-re-dir)
                        (not (member filename-no-dir (list "." "..")))
                        (not (eolp)))
-              (let ((path (dired-utils-get-filename))
+              (let ((path (dired-get-filename nil t))
                     files)
                 (while (and (file-directory-p path)
                             (file-accessible-directory-p path)
-                            (setq files (f-entries path))
+                            (setq files (directory-files path
+                                                         'full
+                                                         directory-files-no-dot-files-regexp))
                             (= 1 (length files)))
                   (setq path (car files)))
                 (if (and (not files)
-                         (equal path (dired-utils-get-filename)))
+                         (equal path (dired-get-filename nil t)))
```

Comment:
TBH I created the "utils" function to avoid these `nil t` arguments everywhere. To me it seems like quite a bad API and just a bit confusing passing some boolean flags.

People usually install multiple dired-hacks packages and the utils package will be pulled in 99% of the cases, so I think it's "worth it" to keep it.

`f` is also present in virtually every user's Emacs since it's depended upon in so many packages.  In general I'm not a huge fan of "removing dependencies" just for the sake of not having them.  I agree that dash here is not necessary because the built-in performs identical function.